distrepos/link_static.py

```diff
@@ -0,0 +1,63 @@
+import typing as t
+from pathlib import Path
+from distrepos.params import Options
+
+
+def link_static_data(options: Options, repo_name: str = "osg") -> t.Tuple[bool, str]:
+    """
+    Utility function to create a symlink to each top-level directory under options.static_root
+    from options.dest_root
+    
+    Args:
+        options: The global options for the run
+        repo_name: The repo to link between dest_root and static_root
+
+    Returns:
+        An (ok, error message) tuple.
+
+    TODO: "osg" repo is essentially hardcoded by the default arg here, might want to specify
+          somewhere in config instead
+    """
+    if not options.static_root:
+        # no static data specified, no op
+        return True, ""
+
+    # This code assumes options.static_root is an absolute path
+    if not Path('/') in options.static_root.parents:
+        return False, f"Static data path must be absolute, got {options.static_root}"
+
+    static_src = options.static_root / repo_name
+    data_dst = options.dest_root / repo_name
+    
+    if not static_src.exists():
+        return False, f"Static data path {static_src} does not exist"
+
+    if not data_dst.exists():
+        # TODO should this be an error instead?
```

Comment:
I think this directory would normally be created by the rsync action; if you're running the link_static action before the rsync action then the dir wouldn't exist the first time around. So no, it shouldn't be an error.